pkg/manager/config/proxy.go

```diff
@@ -24,46 +24,154 @@ import (
 	"go.uber.org/zap"
 )
 
+type CfgType int
+
+const (
+	CfgServer CfgType = iota
+	CfgLog
+)
+
+type OnlineCfgTypes interface {
+	config.ProxyServerOnline | config.LogOnline
+}
+
+type imeta interface {
+	getPrefix() string
+	unmarshal(bytes []byte) (any, error)
+	addToCh(any)
+	getInitial(cfg *config.Config) any
+}
+
+type meta[T OnlineCfgTypes] struct {
+	prefix   string
+	initFunc func(cfg *config.Config) T
+	ch       chan *T
+}
+
+func newMeta[T OnlineCfgTypes](prefix string, initFunc func(cfg *config.Config) T) *meta[T] {
+	return &meta[T]{
+		prefix:   prefix,
+		initFunc: initFunc,
+		ch:       make(chan *T, 1),
+	}
+}
+
+func (m meta[T]) unmarshal(bytes []byte) (any, error) {
+	var t T
+	err := json.Unmarshal(bytes, &t)
+	return &t, err
+}
+
+func (m meta[T]) getPrefix() string {
+	return m.prefix
+}
+
+func (m meta[T]) addToCh(obj any) {
+	m.ch <- obj.(*T)
+}
+
+func (m meta[T]) getInitial(cfg *config.Config) any {
+	return m.initFunc(cfg)
+}
+
+func (e *ConfigManager) initMetas() {
+	e.metas = map[CfgType]imeta{
+		CfgServer: newMeta(pathPrefixProxyServer, func(cfg *config.Config) config.ProxyServerOnline {
+			return cfg.Proxy.ProxyServerOnline
+		}),
+		CfgLog: newMeta(pathPrefixLog, func(cfg *config.Config) config.LogOnline {
+			return cfg.Log.LogOnline
+		}),
+	}
+}
+
 func (e *ConfigManager) watchCfgProxy(ctx context.Context, cfg *config.Config) error {
```

Comment:
We should rename `watchCfgProxy` to something else like `watchConfigField`, `watchConfigItem`, or `watchConfig` maybe.

Reply:
Maybe I need to also change the HTTP API from `config/proxy` to `config/server`. Does anything (such as operator) rely on this API currently?

Reply:
It does use hard coded `API` path. Because `/lib/cli` does not support passing SSL, I have not actuall use `GetRootCmd`. Maybe change API path later.

But I think API names are OK, it is called `proxyServer` in config or so. Reasonable to call it `/config/proxy`.